### DISCUSSION TITLE
My Shlink links doesn't recognize the referrer link when using it in LinkStack

### DISCUSSION THREAD

**Discussion by leadvic** [JulianPrieber/littlelink-custom#743] [Questions]
Hi there,

I'm using Shlink to track locations and where does the click to a link comes from (referrer), for this I added the link of my shlink url into the buttons of Linkstack, but when I check the referrer, I would have expected to see referrer = "linkstack.example.com", but instead it shows as it was a direct connection (it means just going to the url as it is). This is the first time I see this behavior, as my Shlink urls in Instagram, or a custom webpage with ReactJS, all of them show the correct referrer.

I'm asking this here to ask for help in understanding why is this happening, and what could I do in order to have the expected behavior mentioned before, where my referrer would be the url of my linkstack instead of it being considered as a direct connection.

UPDATE: By the way, I realized that the buttons have the issues I just commented above, on the other hand, the social media links (page icons) works perfectly as I would expect related to the referrer information. So, the only issue I have is with the button links.

**Comment by leadvic** [comment]
After investigating more about referrer, I got to understand that this "referrer" or "referer" is in the header of the url, and while inspecting my hosted instance of Linkstack I realized that the links in "My Links" section have the following information:

```html
<a class="button button-website button button-hover icon-hover" rel="noopener noreferrer nofollow" href="https://linkstack.mydomain.com/going/1?https://shlink.mydomain.com/website" target="_blank"><img alt="website" class="icon hvr-icon" src="https://linkstack.mydomain.com/assets/linkstack/icons/website.svg ">Página Web</a>
```
So, the issue I'm facing is in the `rel="noopener noreferrer nofollow"` part of the url, as this avoids the referrer that I need. Just to let yo know, this section of the url is not in the "Page Icons" section, so those have their referrer working properly.

After understanding this issue, I would like to ask, is there some way to add some configuration variable to let the users choose whether they want this function of referrers or not.

**Comment by JulianPrieber** [comment]
Remove "noreferrer" from the rel section for links
